rtengine/iptransform.cc

```diff
@@ -1244,6 +1259,21 @@ void ImProcFunctions::transformGeneral(bool highQuality, Imagefloat *original, I
                 pLCPMap->correctDistortion(x_d, y_d, w2, h2);
             }
 
+            x_d /= params->perspective.camera_scale;
+            y_d /= params->perspective.camera_scale;
+            if (params->perspective.camera_defish) {
+                double const focal_source = params->perspective.camera_focal_length * maxRadius * 0.0462;
```

Comment:
Maybe `0.0462` isn't that magical, but I can't tell from reading the code how you came up with it. A probably good idea was to put a `constexpr double reasonable_name = 0.0462;` into an anonymous namespace at the beginning of the file, or when used only locally, to the start of the function.

Reply:
It's kind of magical as in I don't understand it and figured it out by experiment:
It converts the focal length set by the user into a focal length that makes sense for the current scale of x_d and y_d.
I adjusted it so it works for the rendered checkerboard which has a "physical focal length" of ~8.09mm on a small frame sensor.
It also works for my real 6.5mm lens